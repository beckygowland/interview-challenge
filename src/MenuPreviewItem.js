import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { deleteFoodItem } from './reduxStore';

export class MenuPreviewItem extends React.PureComponent {

    delete = () => {
        const { id, position, deleteFoodItem } = this.props;
        deleteFoodItem(id, position)
    }

    render() {
        const { name, dietaries } = this.props;
        return (
            <li className="item">
                <h2>{name}</h2>
                <p>
                    {dietaries.map((dietary) =>
                        <span key={dietary} className="dietary">{dietary}</span>
                    )}
                </p>
                <button className="remove-item" onClick={this.delete}>x</button>
            </li> 
        );
    }
}

MenuPreviewItem.propTypes = {
    id: PropTypes.number.isRequired,
    position: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    dietaries: PropTypes.array,
    deleteFoodItem: PropTypes.func.isRequired,
};

MenuPreviewItem.defaultProps = {
    dietaries: [],
};

const mapDispatchToProps = {
    deleteFoodItem,
};

export default connect(() => ({}), mapDispatchToProps)(MenuPreviewItem);