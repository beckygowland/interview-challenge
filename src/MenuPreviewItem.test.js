import React from 'react';
import { MenuPreviewItem } from './MenuPreviewItem';
import sinon from 'sinon';

import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('renders', () => {
  const div = shallow(<MenuPreviewItem id={100} name="itemTest" 
        position={0} dietaries={['z', 'y', 'x']} deleteFoodItem={() => {}}/>);
  expect(div).toMatchSnapshot();
  expect(div.find('h2').text()).toBe('itemTest');
  expect(div.find('span').length).toBe(3);
});

it('delete click calls delete prop callback', () => {
  const onDeleteClick = sinon.spy();

  const div = shallow(<MenuPreviewItem id={100} name="itemTest" 
        position={0} dietaries={['z', 'y', 'x']} deleteFoodItem={onDeleteClick}/>);
  div.find('button').simulate('click');
  expect(onDeleteClick.callCount).toBe(1);
  expect(onDeleteClick.args[0][0]).toBe(100);
  expect(onDeleteClick.args[0][1]).toBe(0);
});
