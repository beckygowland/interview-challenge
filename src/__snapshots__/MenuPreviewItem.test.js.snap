// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MenuPreviewItem
    deleteFoodItem={[Function]}
    dietaries={
        Array [
            "z",
            "y",
            "x",
          ]
    }
    id={100}
    name="itemTest"
    position={0}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          itemTest
</h2>,
        <p>
          <span
                    className="dietary"
          >
                    z
          </span>
          <span
                    className="dietary"
          >
                    y
          </span>
          <span
                    className="dietary"
          >
                    x
          </span>
</p>,
        <button
          className="remove-item"
          onClick={[Function]}
>
          x
</button>,
      ],
      "className": "item",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "itemTest",
        },
        "ref": null,
        "rendered": "itemTest",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="dietary"
>
              z
</span>,
            <span
              className="dietary"
>
              y
</span>,
            <span
              className="dietary"
>
              x
</span>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "z",
            "nodeType": "host",
            "props": Object {
              "children": "z",
              "className": "dietary",
            },
            "ref": null,
            "rendered": "z",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "y",
            "nodeType": "host",
            "props": Object {
              "children": "y",
              "className": "dietary",
            },
            "ref": null,
            "rendered": "y",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": "x",
            "nodeType": "host",
            "props": Object {
              "children": "x",
              "className": "dietary",
            },
            "ref": null,
            "rendered": "x",
            "type": "span",
          },
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "x",
          "className": "remove-item",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "x",
        "type": "button",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            itemTest
</h2>,
          <p>
            <span
                        className="dietary"
            >
                        z
            </span>
            <span
                        className="dietary"
            >
                        y
            </span>
            <span
                        className="dietary"
            >
                        x
            </span>
</p>,
          <button
            className="remove-item"
            onClick={[Function]}
>
            x
</button>,
        ],
        "className": "item",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "itemTest",
          },
          "ref": null,
          "rendered": "itemTest",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="dietary"
>
                z
</span>,
              <span
                className="dietary"
>
                y
</span>,
              <span
                className="dietary"
>
                x
</span>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "z",
              "nodeType": "host",
              "props": Object {
                "children": "z",
                "className": "dietary",
              },
              "ref": null,
              "rendered": "z",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "y",
              "nodeType": "host",
              "props": Object {
                "children": "y",
                "className": "dietary",
              },
              "ref": null,
              "rendered": "y",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": "x",
              "nodeType": "host",
              "props": Object {
                "children": "x",
                "className": "dietary",
              },
              "ref": null,
              "rendered": "x",
              "type": "span",
            },
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "x",
            "className": "remove-item",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "x",
          "type": "button",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
